src/averageAges.js

```diff
@@ -55,7 +65,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = people.filter(person =>
+    person.sex === 'f'
+    && people.some(child => child.mother === person.name));
+
+  const children = people.filter(person =>
+    onlyWithSon
+      ? people.some(mother => person.mother === mother.name)
+        && person.sex === 'm'
```

Comment:
if it's a girl, you don’t need to check mother

```suggestion
      ? person.sex === 'm'
        && people.some(mother => person.mother === mother.name)
```